lib/galaxy/datatypes/microarrays.py

```diff
@@ -0,0 +1,159 @@
+    MetadataElement(name="file_type", default="GenePix", desc="File type",
+                    readonly=True, visible=True,
+                    optional=True, no_value="GenePix")
+    MetadataElement(name="block_count", default=1, desc="Number of blocks described in the file",
+                    readonly=True, visible=True,
+                    optional=True, no_value=1)
+    MetadataElement(name="block_type", default=0, desc="Type of block",
+                    readonly=True, visible=True,
+                    optional=True, no_value=0)
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            if dataset.metadata.block_count == 1:
+                dataset.blurb = "%s %s: Format %s, 1 block, %s headers and %s columns" % (dataset.metadata.file_type, dataset.metadata.version_number, dataset.metadata.file_format, dataset.metadata.number_of_optional_header_records, dataset.metadata.number_of_data_columns)
+            else:
+                dataset.blurb = "%s %s: Format %s, %s blocks, %s headers and %s columns" % (dataset.metadata.file_type, dataset.metadata.version_number, dataset.metadata.file_format, dataset.metadata.block_count, dataset.metadata.number_of_optional_header_records, dataset.metadata.number_of_data_columns)
+            dataset.peek = get_file_peek(dataset.file_name)
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disk'
+
+    def get_mime(self):
+        return 'text/plain'
+
+
+@build_sniff_from_prefix
+class Gal(GenericMicroarrayFile):
+    """ Gal File format described at:
+            http://mdc.custhelp.com/app/answers/detail/a_id/18883/#gal
+    """
+
+    edam_format = "format_3829"
+    edam_data = "data_3110"
+    file_ext = "gal"
+
+    def sniff_prefix(self, file_prefix):
+        """
+        Try to guess if the file is a Gal file.
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname('test.gal')
+        >>> Gal().sniff(fname)
+        True
+        >>> fname = get_test_fname('test.gpr')
+        >>> Gal().sniff(fname)
+        False
+        """
+        headers = iter_headers(file_prefix, sep="\t", count=3)
+        found_gal = False
+        found_atf = False
+        for count, line in enumerate(headers):
+            if count == 0:
+                if "ATF" in line[0]:
+                    found_atf = True
+            elif count == 2:
+                if "GenePix ArrayList" in line[0]:
+                    found_gal = True
+            count += 1
+        return found_gal and found_atf
+
+    def set_meta(self, dataset, **kwd):
+        """
+        Set metadata for Gal file.
+        """
+        super(Gal, self).set_meta(dataset, **kwd)
+        headers = iter_headers(dataset.file_name, sep="\t", count=5)
+        for count, line in enumerate(headers):
+            if count == 0:
+                dataset.metadata.file_format = str(line[0])
```

Comment:
I think you can remove the cast to `str` here and below (6 times).

Reply:
Done

Reply:
@nsoranzo any idea why the get code and test fails?.

Reply:
It's an old issue that has been fixed recently in https://github.com/galaxyproject/galaxy/pull/7446 , which was merged after you opened this pull request. Bottom line is: don't worry about it.